rts/Sim/MoveTypes/MoveDefHandler.cpp

```diff
@@ -153,6 +154,16 @@ void MoveDefHandler::Init(LuaParser* defsParser)
 }
 
 
+void MoveDefHandler::PostSimInit() {
+	// Pathing system is initialized later and the movedefs may need to adjust their behaviour to align with
+	// capabilities of the pathing system.
+	std::for_each(moveDefs.begin(), moveDefs.end(), [](MoveDef& md){
+		md.allowDirectionalPathing &= pathManager->AllowDirectionalPathing();
```

Comment:
What happens if I use QTPFS and give a raw move order to tell a unit to move across a downward slope? Or if the only available path is downwards? Does the unit slow down?

Reply:
The unit's speed is directional regardless of the pathing setting. This has been the case for at least 8 years.

Reply:
What about this check removed by this PR?

https://github.com/beyond-all-reason/spring/blob/44f0a4d2376dcda6fd21de512976e1507fe54bf0/rts/Sim/MoveTypes/MoveMath/GroundMoveMath.cpp#L31-L36

![image](https://github.com/beyond-all-reason/spring/assets/2573076/61d90e0b-fa4c-4b9d-825c-f807afffa056)

Reply:
Ah, well spotted. Well my 8 years comment stands corrected.

Reply:
If there is no longer way to make units slow down downhill then it's technically a regression, but it looks like every known game used directional movement anyway so nobody is affected. So I think there is nothing to do for now.

Reply:
I'm fine to add that option back in. Maybe it would make for a good 'start-here' type ticket?

Reply:
I think ideally there would be a good default for movement (and IMO the current fits that description) and, down the line, a way to fully control it. Only having a "downhill slowdown is same as uphill" checkbox is super bland and would be the engine only allowing implementing specific game design.

For example, what if I want a simple but custom slope slowdown formula that goes like this? 
* flat < 10°: 0% speed change
* upward ramp, > 10°, < 40°: -33% slowdown
* upward steep ramp, > 40° (as long as within maxslope): -66% slowdown
* downward ramp, > 10°, < 40°: 0% change for movedef INFANTRY, +25% speedup for movedef VEHICLE
* downward ramp, > 40°: -50% slowdown for movedef INFANTRY (gotta tread carefully), +100% speedup for movedef VEHICLE (rollin')

Right now there doesn't seem to be a nice way for a game to control movement speed like this, and the bool option for the previous non-directional formula wouldn't help much either. The questions would then be:
* what would be a good interface to define such formula?
* how well could pathfinders take any custom formulas into account?

It's a scary task (big, but also a lot of opportunity to silently screw things up) and the default seems decent enough.